caravel/assets/javascripts/SqlLab/components/SqlEditor.jsx

```diff
@@ -149,17 +149,36 @@ class SqlEditor extends React.Component {
   }
 
   render() {
-    let runButtons = (
-      <ButtonGroup bsSize="small" className="inline m-r-5 pull-left">
+    let runButtons = [];
+    if (this.props.database && this.props.database.allow_run_sync) {
+      runButtons.push(
         <Button
           bsSize="small"
           bsStyle="primary"
           style={{ width: '100px' }}
-          onClick={this.runQuery.bind(this)}
+          onClick={this.runQuery.bind(this, false)}
           disabled={!(this.props.queryEditor.dbId)}
         >
           <i className="fa fa-table" /> Run Query
         </Button>
+      );
+    }
+    if (this.props.database && this.props.database.allow_run_async) {
+      runButtons.push(
```

Comment:
would be nice to DRY this up. could make a `<RunButton />` component to encapsulate the logic:

```
<RunButton
  async={this.props.database && this.props.database.allow_run_async}
  sync={this.props.database && this.props.database.allow_run_sync}
  isRunning={this.props.latestQuery && this.props.latestQuery.state === 'running'}
/>
```